src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -10648,202 +10666,1371 @@ static void Main()
-
-            var actualMembers = GetProperties(comp, "C").ToTestDisplayStrings();
-            var expectedMembers = new[]
-            {
-                "System.Type C.EqualityContract { get; }",
-                "System.Object C.Q { get; init; }",
-                "System.Object C.P { get; }",
-                "System.Object C.P { get; }",
-            };
-            AssertEx.Equal(expectedMembers, actualMembers);
-        }
+    internal static bool Report(string s) { System.Console.WriteLine(s); return false; }
 
-        [WorkItem(44692, "https://github.com/dotnet/roslyn/issues/44692")]
-        [Fact]
-        public void DuplicateProperty_02()
+    protected abstract System.Type EqualityContract { get; }
+}
+record B : A
+{
+    protected override System.Type EqualityContract
+    {
+        get
         {
-            var src =
-@"record C(object P, object Q)
+            Report(""B.EqualityContract"");
+            return typeof(B);
+        }
+    }
+}
+class Program
 {
-    public object P { get; }
-    public int P { get; }
-    public int Q { get; }
-    public object Q { get; }
-}";
-            var comp = CreateCompilation(src);
-            comp.VerifyDiagnostics(
-                // (1,27): error CS8866: Record member 'C.Q' must be a readable instance property of type 'object' to match positional parameter 'Q'.
-                // record C(object P, object Q)
-                Diagnostic(ErrorCode.ERR_BadRecordMemberForPositionalParameter, "Q").WithArguments("C.Q", "object", "Q").WithLocation(1, 27),
-                // (4,16): error CS0102: The type 'C' already contains a definition for 'P'
-                //     public int P { get; }
-                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "P").WithArguments("C", "P").WithLocation(4, 16),
-                // (6,19): error CS0102: The type 'C' already contains a definition for 'Q'
-                //     public object Q { get; }
-                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "Q").WithArguments("C", "Q").WithLocation(6, 19));
+    static void Main()
+    {
+        A a1 = new B();
+        A a2 = new B();
 
-            var actualMembers = GetProperties(comp, "C").ToTestDisplayStrings();
-            var expectedMembers = new[]
-            {
-                "System.Type C.EqualityContract { get; }",
-                "System.Object C.P { get; }",
-                "System.Int32 C.P { get; }",
-                "System.Int32 C.Q { get; }",
-                "System.Object C.Q { get; }",
-            };
-            AssertEx.Equal(expectedMembers, actualMembers);
+        System.Console.WriteLine(a1.Equals(a2));
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput:
+@"
+B.EqualityContract
```

Comment:
I didn't understand why there are two calls to EqualityContract

Reply:
> I didn't understand why there are two calls to EqualityContract

From specification for record equals method:
If there is a base record type, the value of base.Equals(other) (a non-virtual call to public virtual bool Equals(Base? other)); otherwise the value of EqualityContract == other.EqualityContract.

---
In reply to: [452484887](https://github.com/dotnet/roslyn/pull/45831#discussion_r452484887) [](ancestors = 452484887)